App crashes with openapi file generation for credentialsApi fails

If there are issues in generating the openapi file, then the application is fully crashing. 
We must log this error and still continue with the application startup.

14:49:55.745 [main] INFO  o.e.x.common.openapi.OpenApiUtil []- Download openapi client:D:\Users\s00828547\IdeaProjects\forked_clean\osc\openapi\openapi-generator-cli.jar from URL:https://repo1.maven.org/maven2/org/openapitools/openapi-generator-cli/6.5.0/openapi-generator-cli-6.5.0.jar start.
14:50:17.326 [main] ERROR o.e.x.m.c.CredentialOpenApiGenerator []- credentialApi html file:flexibleEngine_variables_credentialApi.html creation failed.
java.net.ConnectException: Connection timed out: connect
14:50:17.379 [main] ERROR o.s.boot.SpringApplication []- Application run failed
java.lang.IllegalStateException: Failed to execute ApplicationRunner
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:765)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:752)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:319)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1306)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1295)
	at org.eclipse.xpanse.runtime.XpanseApplication.main(XpanseApplication.java:30)
Caused by: org.eclipse.xpanse.modules.models.common.exceptions.OpenApiFileGenerationException: credentialApi html file creation failed: Connection timed out: connect
	at org.eclipse.xpanse.modules.credential.CredentialOpenApiGenerator.createCredentialApi(CredentialOpenApiGenerator.java:161)
	at org.eclipse.xpanse.modules.credential.CredentialOpenApiGenerator.lambda$run$1(CredentialOpenApiGenerator.java:97)
	at java.base/java.util.HashMap$KeySet.forEach(HashMap.java:1008)
	at org.eclipse.xpanse.modules.credential.CredentialOpenApiGenerator.run(CredentialOpenApiGenerator.java:94)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:762)
	... 5 common frames omitted

--- a/modules/credential/src/main/java/org/eclipse/xpanse/modules/credential/CredentialOpenApiGenerator.java
+++ b/modules/credential/src/main/java/org/eclipse/xpanse/modules/credential/CredentialOpenApiGenerator.java
@@ -31,8 +31,8 @@ import org.eclipse.xpanse.modules.orchestrator.OrchestratorPlugin;
 import org.eclipse.xpanse.modules.orchestrator.PluginManager;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
-import org.springframework.boot.ApplicationArguments;
-import org.springframework.boot.ApplicationRunner;
+import org.springframework.boot.context.event.ApplicationStartedEvent;
+import org.springframework.context.ApplicationListener;
 import org.springframework.stereotype.Component;
 
 /**
@@ -40,12 +40,13 @@ import org.springframework.stereotype.Component;
  */
 @Slf4j
 @Component
-public class CredentialOpenApiGenerator implements ApplicationRunner {
+public class CredentialOpenApiGenerator implements ApplicationListener<ApplicationStartedEvent> {
 
     private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
     private final String appVersion;
     private final OpenApiUtil openApiUtil;
     private final PluginManager pluginManager;
+
 
     /**
      * Constructor of CredentialApiUtil.
@@ -54,7 +55,7 @@ public class CredentialOpenApiGenerator implements ApplicationRunner {
     public CredentialOpenApiGenerator(PluginManager pluginManager,
                                       OpenApiUtil openApiUtil,
                                       @Value("${app.version:1.0.0}")
-                                      String appVersion) {
+                                              String appVersion) {
         this.pluginManager = pluginManager;
         this.openApiUtil = openApiUtil;
         this.appVersion = appVersion;
@@ -78,14 +79,12 @@ public class CredentialOpenApiGenerator implements ApplicationRunner {
         return openApiUtil.getOpenApiWorkdir();
     }
 
-
     @Override
-    public void run(ApplicationArguments args) {
+    public void onApplicationEvent(ApplicationStartedEvent event) {
         Map<Csp, OrchestratorPlugin> cspOrchestratorPluginMap = pluginManager.getPluginsMap();
-        for (Csp csp : cspOrchestratorPluginMap.keySet()) {
-            OrchestratorPlugin orchestratorPlugin = cspOrchestratorPluginMap.get(csp);
+        for (Map.Entry<Csp, OrchestratorPlugin> entry : cspOrchestratorPluginMap.entrySet()) {
             List<AbstractCredentialInfo> credentialDefinitions =
-                    orchestratorPlugin.getCredentialDefinitions();
+                    entry.getValue().getCredentialDefinitions();
             if (Objects.nonNull(credentialDefinitions)) {
                 Map<CredentialType, AbstractCredentialInfo> typeCredentialInfoMap =
                         credentialDefinitions.stream().filter(Objects::nonNull)
@@ -98,10 +97,11 @@ public class CredentialOpenApiGenerator implements ApplicationRunner {
                 });
             } else {
                 log.info("Not found credential definition of the cloud service provider:{}",
-                        csp.toValue());
+                        entry.getKey().toValue());
             }
         }
     }
+
 
     /**
      * create credentialApi for plugins.

--- a/runtime/src/test/java/org/eclipse/xpanse/runtime/modules/credential/CredentialOpenApiGeneratorTest.java
+++ b/runtime/src/test/java/org/eclipse/xpanse/runtime/modules/credential/CredentialOpenApiGeneratorTest.java
@@ -19,6 +19,7 @@ import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
 import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
 
@@ -26,6 +27,7 @@ import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
  * Test of CredentialOpenApiGenerator.
  */
 @ExtendWith(SpringExtension.class)
+@ActiveProfiles("default")
 @SpringBootTest(classes = {XpanseApplication.class, CaffeineCredentialCacheManager.class,
         CredentialsStore.class,
         ServletUriComponentsBuilder.class, OpenApiUtil.class, PluginManager.class})
@@ -33,10 +35,6 @@ class CredentialOpenApiGeneratorTest {
 
     @Autowired
     private CredentialOpenApiGenerator credentialOpenApiGenerator;
-
-    @Autowired
-    private OpenApiUtil openApiUtil;
-
 
     @Test
     void testGetServiceUrl() {
@@ -48,7 +46,7 @@ class CredentialOpenApiGeneratorTest {
 
     @Test
     void testRun() {
-        credentialOpenApiGenerator.run(null);
+        credentialOpenApiGenerator.onApplicationEvent(null);
     }
 
     @Test

